src/syscheckd/src/create_db.c

```diff
@@ -136,32 +243,72 @@ static void transaction_callback(ReturnTypeCallback resultType, const cJSON* res
-            event_data->evt_data->type = FIM_ADD;
+            txn_context->evt_data->type = FIM_ADD;
             break;
 
         case MODIFIED:
-            event_data->evt_data->type = FIM_MODIFICATION;
+            txn_context->evt_data->type = FIM_MODIFICATION;
             break;
 
         case DELETED:
             if (configuration->options & CHECK_SEECHANGES) {
                 fim_diff_process_delete_file(path);
             }
-            event_data->evt_data->type = FIM_DELETE;
-            event_data->db_full = false;
+
+            txn_context->evt_data->type = FIM_DELETE;
+            txn_context->db_full = false;
 
             break;
 
         case MAX_ROWS:
-            event_data->db_full = true;
+            txn_context->db_full = true;
             mdebug1("Couldn't insert '%s' entry into DB. The DB is full, please check your configuration.", path);
 
         // Fallthrough
         default:
             goto end;
     }
 
-    json_event = fim_dbsync_json_event(path, diff, dbsync_event, configuration, event_data->evt_data);
+    json_event = cJSON_CreateObject();
+    data = cJSON_CreateObject();
+
+    if (json_event == NULL || data == NULL) {
+        goto end;
+    }
+
+    cJSON_AddStringToObject(json_event, "type", "event");
+    cJSON_AddItemToObject(json_event, "data", data);
+
+    cJSON_AddStringToObject(data, "path", path);
+    cJSON_AddNumberToObject(data, "version", 2.0);
+    cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[txn_context->evt_data->mode]);
+    cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[txn_context->evt_data->type]);
+    //cJSON_AddNumberToObject(data, "timestamp", time(NULL)); This should be the last_event field
+
+    if (resultType == DELETED || txn_context->latest_entry == NULL) {
+        // We need to add the `type` field to the attributes JSON. This avoid modifying the dbsync event.
+        cJSON *attributes = cJSON_Duplicate(dbsync_event, 1);
+        cJSON_AddStringToObject(attributes, "type", "file");
+        cJSON_AddItemToObject(data, "attributes", attributes);
+    } else {
+        cJSON_AddItemToObject(data, "attributes", fim_attributes_json(txn_context->latest_entry->file_entry.data));
+        old_data = cJSON_GetObjectItem(dbsync_event, "old");
+
+        if (old_data) {
+            old_attributes = cJSON_CreateObject();
+            changed_attributes = cJSON_CreateArray();
+            cJSON_AddItemToObject(data, "old_attributes", old_attributes);
+            cJSON_AddItemToObject(data, "changed_attributes", changed_attributes);
+            fim_calculate_dbsync_difference(txn_context->latest_entry->file_entry.data,
+                                            old_data,
+                                            old_attributes,
+                                            changed_attributes);
+        }
+    }
+
+    if (configuration->tag != NULL) {
+        cJSON_AddStringToObject(data, "tags", configuration->tag);
+    }
 
```

Comment:
In the transaction_callback, the content_changes tag, must be added to the json using the diff
```
        if (diff != NULL) {
            cJSON_AddStringToObject(data, "content_changes", diff);
        }
```